### PR TITLE
Python bindings

### PR DESCRIPTION
**Pull request by troels-im**
This PR introduces Python bindings that can be used to expose muddle to Python. This will enable the agent team to quickly build P2P networks.

### CODE REVIEW

File: libs/python/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
+project(fetch-python)
+
+# CMake Configuration
+include(${FETCH_ROOT_CMAKE_DIR}/BuildTools.cmake)
+
+# Compiler Configuration
+setup_compiler()
+
+# if (FETCH_ENABLE_PYTHON_BINDINGS)
```
Comment (baykaner) [inline libs/python/CMakeLists.txt]: commented out if statement is redundant? remove it?

File: libs/python/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+target_link_libraries(fetch-python
+                      PRIVATE fetch-math
+                              fetch-crypto
+                              fetch-muddle
+                              fetch-core)
+
+# make the name match the bindings
+set_target_properties(fetch-python PROPERTIES OUTPUT_NAME "fetch")
+
+# endif (FETCH_ENABLE_PYTHON_BINDINGS)
```
Comment (baykaner) [inline libs/python/CMakeLists.txt]: commented out end if

File: libs/python/setup.py.in [hunk 3]
```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+
```
Comment (baykaner) [inline libs/python/setup.py.in]: is this file supposed to be here? i'm not really sure what a .py.in filetype is. this looks like an ordinary python file

**Review by baykaner — CHANGES_REQUESTED** [review]

File: libs/python/CMakeLists.txt (hunk 1 above)
Comment (baykaner) [reply libs/python/CMakeLists.txt]: looks like FETCH_ENABLE_PYTHON_BINDINGS is still in cmakelists file, why not just uncomment this?

File: libs/python/setup.py.in (hunk 3 above)
Comment (baykaner) [reply libs/python/setup.py.in]: i think this file is auto generated by pyconfigure?

File: libs/python/CMakeLists.txt (hunk 1 above)
Comment (troels-im) [reply libs/python/CMakeLists.txt]:
```suggestion
```

File: libs/python/CMakeLists.txt (hunk 2 above)
Comment (troels-im) [reply libs/python/CMakeLists.txt]:
```suggestion
```

**Review by baykaner — DISMISSED** [review]

File: libs/python/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#
+# F E T C H   P Y T H O N   B I N D I N G S   L I B R A R Y
+#
+cmake_minimum_required(VERSION 3.10 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
```
Comment (ejfitzgerald) [inline libs/python/CMakeLists.txt]: Redundant line

**Review by ejfitzgerald — CHANGES_REQUESTED** [review]
There are some changes here, but I don't see any of the bindings for Muddle which is sort of what I was expected. Is that not the intention here?

File: libs/python/CMakeLists.txt (hunk 4 above)
Comment (troels-im) [reply libs/python/CMakeLists.txt]:
```suggestion
```

**Review by ejfitzgerald — APPROVED** [review]

**Pull request merged**